src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,31 @@
+package seedu.addressbook.data.person;
+
+/**
+ * 
+ * Stores the details of the contacts 
+ *
+ */
+public class Contact {
+
+    public String value;
+    protected boolean isPrivate;
+
+    public Contact() {
```

Comment:
I think you have missed the concept of Inheritance here a bit. The super class in the specific scenario is the Contact class. So a more meaningful implementation is push a general constructor to the Contant class and call the super() from the sub class Hint. think of a constructor as Contact(String value, Boolean isPrivate)